src/record_ts.cpp

```diff
@@ -471,6 +475,41 @@ void RecordThreadState::barrier() {
     return m_internal->barrier();
 }
 
+void RecordThreadState::notify_opaque_width(uint32_t index,
+                                              uint32_t width_index) {
+    if (!paused()) {
+        uint32_t start = m_recording.dependencies.size();
+        Variable *v1 = jitc_var(index);
+        Variable *v2 = jitc_var(width_index);
+        add_in_param(v1->data, (VarType) v1->type);
+        add_out_param(v2->data, VarType::UInt32);
+        uint32_t end = m_recording.dependencies.size();
+
+        Operation op;
+        op.type             = OpType::OpaqueWidth;
+        op.dependency_range = std::pair(start, end);
+        m_recording.operations.push_back(op);
+    }
+}
+
+int Recording::replay_opaque_width(Operation &op){
```

Comment:
Very minor: Missing space between `){`, extra newline after start of the function